setup.py

```diff
@@ -7,19 +7,24 @@
     # pip >=20
     from pip._internal.network.session import PipSession
     from pip._internal.req import parse_requirements
+    install_requires = parse_requirements('requirements.txt',session=PipSession())
+    dependencies = [str(package.requirement) for package in install_requires]
 except ImportError:
     try:
         # 10.0.0 <= pip <= 19.3.1
         from pip._internal.download import PipSession
-        from pip._internal.req import parse_requirements
+        from pip._internal.req import parse_requirements       
     except ImportError:
         # pip <= 9.0.3
         from pip.download import PipSession
         from pip.req import parse_requirements
 
+    install_requires = parse_requirements('requirements.txt',session=PipSession())
```

Comment:
Line 22-23 is the same as 10-11, define it in one place (above try should be fine). We should always follow DRY

Reply:
But 'parse_requirements' is also pip version dependent. How can I place it above try? Can you please elaborate?

Reply:
We can keep the one at line 22-23 then. That'll work

Reply:
move it out of the except block

Reply:
Did you mean to say `package.req` (line 22) will work instead of `package.requirement` (line 11)? But it gave errors when i tried.

Reply:
What i have understood is that, both are two different lines. since parse_requirements module has changed across versions and thus for pip versions greater than 20, it allows `package.requirement`(Line 11) where as for lower versions it allows `package.req` (Line 23).
When i moved those lines out of except block, and made a build i got [error.]( https://travis-ci.org/github/priyanshunayan/hydra-python-agent)

Reply:
Cool!